db/upgrade.php

```diff
@@ -60,6 +60,26 @@ function xmldb_ojt_upgrade($oldversion) {
         upgrade_mod_savepoint(true, 2016031400, 'ojt');
     }
 
+    if ($oldversion < 2021041400) {
+        $table = new xmldb_table('ojt_topic');
+        $field = new xmldb_field('position', XMLDB_TYPE_INTEGER, '4', null, null, null, '0', 'allowcomments');
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        $table = new xmldb_table('ojt_topic_item');
+        $field = new xmldb_field('position', XMLDB_TYPE_INTEGER, '4', null, null, null, '0', 'allowselffileuploads');
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        $ojtrs = $DB->get_recordset('ojt');
+        foreach ($ojtrs as $ojt) {
+            ojt_reorder_topics($ojt);
```

Comment:
This errors out on upgrade:
```
-->mod_ojt
Default exception handler: Exception - Call to undefined function ojt_reorder_topics() Debug: 
Error code: generalexceptionmessage
* line 78 of /mod/ojt/db/upgrade.php: Error thrown
* line 836 of /lib/upgradelib.php: call to xmldb_ojt_upgrade()
* line 530 of /lib/upgradelib.php: call to upgrade_plugins_modules()
* line 1939 of /lib/upgradelib.php: call to upgrade_plugins()
* line 196 of /admin/cli/upgrade.php: call to upgrade_noncore()
```

May I suggest adding something like this at the top of the upgrade block:
```php
require_once($CFG->dirroot . '/mod/ojt/locallib.php');
```